packages/medusa/src/api/routes/store/products/get-product.ts

```diff
@@ -76,6 +83,14 @@ export default async (req, res) => {
     relations: defaultStoreProductsRelations,
   })
 
+  let sales_channel_id = validated.sales_channel_id
+  const featureFlagRouter: FlagRouter = req.scope.resolve("featureFlagRouter")
+  if (featureFlagRouter.isFeatureEnabled(PublishableAPIKeysFeatureFlag.key)) {
+    if (req.publishableApiKeyScopes?.sales_channel_id.length === 1) {
+      sales_channel_id = req.publishableApiKeyScopes.sales_channel_id[0]
+    }
+  }
```

Comment:
thought: think the middleware @fPolic wrote already does this

Reply:
not 100%, What the middleware @fPolic wrote does is it will fetch all sales channels and set an array since a publishable key can have many sales channels defined. In the store we want a single sales channel, hence I'm only setting the sales channel if a single channel is defined here.

You are right that we should default the default sales channel if no other channel is defined. How do you think we should gracefully handle an api key with multiple channels?

Reply:
Sorry I misread this code the first time around... 

In the case where a PK works for more than one channel, I have a few ideas we could consider:

1. Throw an error and force consumer to be specific about the sales channel.
2. Default to `channels[0]` like you are doing now.
3. Add a channel availability field to the response E.g.:
  ```
  variant.sales_channel_availability = [{
    channel_id: "",
    channel_name: "",
    available_quantity: xyz,
  }, ...]
  ```

I think the best approach, for now, is to go with 2.